### PR TITLE
Update License struct and add new RepositoryLicense struct

### LINKED ISSUE #471: Licenses struct is out of date
The `Licenses` struct has diverged from the latest returned by the GitHub API (https://developer.github.com/v3/licenses/) and should be updated to reflect the latest state.

### PR DESCRIPTION
**Pull request by nmiyake**
Update structs to reflect the new reponses returned from GitHub.

Fixes #471

### CODE REVIEW

File: github/licenses.go [hunk 1]
```diff
@@ -10,23 +10,49 @@ import "fmt"
 // LicensesService handles communication with the license related
 // methods of the GitHub API.
 //
-// GitHub API docs: http://developer.github.com/v3/pulls/
+// GitHub API docs: http://developer.github.com/v3/licenses/
```
Comment (shurcooL) [inline github/licenses.go]: Mind changing the scheme to https here?

**Comment by shurcooL** [comment]
LGTM. Thanks for reporting and fixing this issue @nmiyake!

**Comment by nmiyake** [comment]
Great, thanks @shurcooL . Any ETA on when it might get merged?

**Comment by shurcooL** [comment]
As soon as @gmlewis or @willnorris or someone else reviews it and gives it a second LGTM.

File: github/licenses.go [hunk 2]
```diff
@@ -10,23 +10,49 @@ import "fmt"
...
+	SHA         *string `json:"sha,omitempty"`
+	Size        *int    `json:"size,omitempty"`
+	URL         *string `json:"url,omitempty"`
+	HTMLURL     *string `json:"html_url,omitempty"`
+	GitURL      *string `json:"git_url,omitempty"`
+	DownloadURL *string `json:"download_url,omitempty"`
+	Type        *string `json:"type,omitempty"`
+	Content     *string `json:"content,omitempty"`
+	Encoding    *string `json:"encoding,omitempty"`
+	Links       *struct {
```
Comment (willnorris) [inline github/licenses.go]: The Links structure simply duplicates the various *URL fields.  I don't think we've included it on any other structs in go-github, so probably doesn't make sense here either.

Comment (nmiyake) [inline github/licenses.go]: Yes, the only other place I see it is in the `Feeds` struct and it has more pertinent information there -- fine with removing this. Will do so and update PR.

**Comment by nmiyake** [comment]
Updated PR to address feedback (removed the `_links` field)

**Pull request merged**